qt/scientific_interfaces/ISISReflectometry/Reduction/LookupTable.cpp

```diff
@@ -0,0 +1,109 @@
+  // which will be used for everything where a specific match is not found
+  return findWildcardLookupRow();
+}
+
+boost::optional<LookupRow> LookupTable::searchByTheta(std::vector<LookupRow> lookupRows,
+                                                      boost::optional<double> const &thetaAngle,
+                                                      double tolerance) const {
+  // TODO We may get multiple matches if the title matches multiple regexes. If one regex is empty then we
+  // can discard it. If we get multiple non-empty regex matches it is an error.
+  auto match =
+      std::find_if(lookupRows.cbegin(), lookupRows.cend(), [thetaAngle, tolerance](LookupRow const &candiate) -> bool {
+        return !candiate.isWildcard() && equalWithinTolerance(*thetaAngle, candiate.thetaOrWildcard().get(), tolerance);
+      });
+  if (match == lookupRows.cend())
+    return boost::none;
+  else
+    return *match;
+}
+
+std::vector<LookupRow> LookupTable::findMatchingRegexes(std::string const &title) const {
+  auto results = std::vector<LookupRow>();
+  std::copy_if(m_lookupRows.cbegin(), m_lookupRows.cend(), std::back_inserter(results),
+               [&title](auto const &candidate) {
+                 return candidate.titleMatcher() && boost::regex_search(title, candidate.titleMatcher().get());
+               });
+  return results;
+}
+
+std::vector<LookupRow> LookupTable::findEmptyRegexes() const {
+  auto results = std::vector<LookupRow>();
+  std::copy_if(m_lookupRows.cbegin(), m_lookupRows.cend(), std::back_inserter(results),
+               [](auto const &candidate) { return !candidate.titleMatcher(); });
+  return results;
+}
+
+std::vector<LookupRow> LookupTable::searchByTitle(Row const &row) const {
+  if (!row.getParent() || row.getParent()->name().empty()) {
+    // No titles for us to check against, so skip filtering
+    return m_lookupRows;
+  }
+
+  auto const &title = row.getParent()->name();
+  auto results = findMatchingRegexes(title);
+  return results;
+}
+
+boost::optional<LookupRow> LookupTable::findWildcardLookupRow() const {
+  auto match = std::find_if(m_lookupRows.cbegin(), m_lookupRows.cend(),
+                            [](LookupRow const &candidate) -> bool { return candidate.isWildcard(); });
+  if (match == m_lookupRows.cend())
+    return boost::none;
+  else
+    return *match;
+}
+
+std::vector<LookupRow::ValueArray> LookupTable::toValueArray() const {
+  auto result = std::vector<LookupRow::ValueArray>();
+  std::transform(m_lookupRows.cbegin(), m_lookupRows.cend(), std::back_inserter(result),
+                 [](auto const &lookupRow) { return lookupRowToArray(lookupRow); });
+  return result;
+};
+
+bool operator==(LookupTable const &lhs, LookupTable const &rhs) { return lhs.m_lookupRows == rhs.m_lookupRows; }
+bool operator!=(LookupTable const &lhs, LookupTable const &rhs) { return !(lhs == rhs); }
```

Comment:
So I looked this up as the current way looked wrong to me (despite writing it originally), apparently the idiomatic way of doing this is 
```c++
return !operator==(lhs, rhs);
```